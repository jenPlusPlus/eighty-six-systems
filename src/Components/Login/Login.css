.login-form-container {
  background-color: #33202A;
  width: 40%;
  height: 65%;
  margin: auto;
  border-radius: 10px;
  position: relative;
  top: 12%; }

.login-form {
  height: 100%;
  padding: 10px;
  display: grid;
  grid-template-columns: auto auto auto;
  grid-template-rows: auto auto auto auto auto;
  grid-column-gap: 5px;
  grid-row-gap: 5px;
  border-radius: 10px; }

.number-button, .login-submit {
  color: #6D576A;
  border-radius: 5px;
  background-color: #EFEFF0;
  border: none;
  font-size: 16px; }

.button-1 {
  grid-column: 1 / 2;
  grid-row: 1 / 2; }

.button-2 {
  grid-column: 2 / 3;
  grid-row: 1 / 2; }

.button-3 {
  grid-column: 3 / 4;
  grid-row: 1 / 2; }

.button-4 {
  grid-column: 1 / 2;
  grid-row: 2 / 3; }

.button-5 {
  grid-column: 2 / 3;
  grid-row: 2 / 3; }

.button-6 {
  grid-column: 3 / 4;
  grid-row: 2 / 3; }

.button-7 {
  grid-column: 1 / 2;
  grid-row: 3 / 4; }

.button-8 {
  grid-column: 2 / 3;
  grid-row: 3 / 4; }

.button-9 {
  grid-column: 3 / 4;
  grid-row: 3 / 4; }

.button-0 {
  grid-column: 2 / 3;
  grid-row: 4 / 5; }

.login-submit {
  grid-row: 5 / 6;
  grid-column: 1 / span 3; }
