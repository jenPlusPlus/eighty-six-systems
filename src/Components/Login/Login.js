import React, { Component } from 'react';
import firebase from './../../firebase.js';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      systemLoginCode: '0000',
      userLoginCode: ''
    };
  }

  updateState(value) {
    this.setState({
      userLoginCode: this.state.userLoginCode.concat(value)
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.state.systemLoginCode === this.state.userLoginCode) {
      const usersRef = firebase.database().ref('users');
      usersRef.orderByChild("loginCode").equalTo(this.state.userLoginCode)
        .once('value', snapshot => {
          const currentUser = Object.entries(snapshot.val());
          this.props.loginUser(currentUser[0][1]);
          alert(`Welcome, ${currentUser[0][1].name}`);
        });
    } else {
      alert('LOGIN CODE INCORRECT!');
    }

    this.setState({
      userLoginCode: ''
    });



    // this.props.loginUser();
  }

  render() {
    return (
      <div className="login-form-container">
        <div className="login-form">
          <button className="login-button"
            onClick={() => this.updateState("1")}>1</button>
          <button className="login-button"
            onClick={() => this.updateState("2")}>2</button>
          <button className="login-button"
            onClick={() => this.updateState("3")}>3</button>
          <button className="login-button"
            onClick={() => this.updateState("4")}>4</button>
          <button className="login-button"
            onClick={() => this.updateState("5")}>5</button>
          <button className="login-button"
            onClick={() => this.updateState("6")}>6</button>
          <button className="login-button"
            onClick={() => this.updateState("7")}>7</button>
          <button className="login-button"
            onClick={() => this.updateState("8")}>8</button>
          <button className="login-button"
            onClick={() => this.updateState("9")}>9</button>
          <button className="login-button"
            onClick={() => this.updateState("0")}>0</button>
          <button className="login-submit"
            onClick={(event) => this.handleSubmit(event)}>Login</button>
        </div>
      </div>
    );
  }
}

export default Login;
