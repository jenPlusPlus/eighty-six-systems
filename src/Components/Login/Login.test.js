import { shallow } from 'enzyme';
import Login from './Login';
import ReactDOM from 'react-dom';
import React from 'react';

describe('Login snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Login
      currentUser={{name: 'Jen', KitchenViewCode: '0000'}}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
