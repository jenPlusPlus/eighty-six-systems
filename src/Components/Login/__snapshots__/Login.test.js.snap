// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    currentUser={
        Object {
            "KitchenViewCode": "0000",
            "name": "Jen",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="login-form"
>
          <button
                    className="number-button button-1"
                    onClick={[Function]}
          >
                    1
          </button>
          <button
                    className="number-button button-2"
                    onClick={[Function]}
          >
                    2
          </button>
          <button
                    className="number-button button-3"
                    onClick={[Function]}
          >
                    3
          </button>
          <button
                    className="number-button button-4"
                    onClick={[Function]}
          >
                    4
          </button>
          <button
                    className="number-button button-5"
                    onClick={[Function]}
          >
                    5
          </button>
          <button
                    className="number-button button-6"
                    onClick={[Function]}
          >
                    6
          </button>
          <button
                    className="number-button button-7"
                    onClick={[Function]}
          >
                    7
          </button>
          <button
                    className="number-button button-8"
                    onClick={[Function]}
          >
                    8
          </button>
          <button
                    className="number-button button-9"
                    onClick={[Function]}
          >
                    9
          </button>
          <button
                    className="number-button button-0"
                    onClick={[Function]}
          >
                    0
          </button>
          <button
                    className="login-submit"
                    onClick={[Function]}
          >
                    Login
          </button>
</div>,
        undefined,
      ],
      "className": "login-form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="number-button button-1"
              onClick={[Function]}
>
              1
</button>,
            <button
              className="number-button button-2"
              onClick={[Function]}
>
              2
</button>,
            <button
              className="number-button button-3"
              onClick={[Function]}
>
              3
</button>,
            <button
              className="number-button button-4"
              onClick={[Function]}
>
              4
</button>,
            <button
              className="number-button button-5"
              onClick={[Function]}
>
              5
</button>,
            <button
              className="number-button button-6"
              onClick={[Function]}
>
              6
</button>,
            <button
              className="number-button button-7"
              onClick={[Function]}
>
              7
</button>,
            <button
              className="number-button button-8"
              onClick={[Function]}
>
              8
</button>,
            <button
              className="number-button button-9"
              onClick={[Function]}
>
              9
</button>,
            <button
              className="number-button button-0"
              onClick={[Function]}
>
              0
</button>,
            <button
              className="login-submit"
              onClick={[Function]}
>
              Login
</button>,
          ],
          "className": "login-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
              "className": "number-button button-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "1",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "className": "number-button button-2",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "2",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3",
              "className": "number-button button-3",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "3",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4",
              "className": "number-button button-4",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "4",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5",
              "className": "number-button button-5",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "5",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "6",
              "className": "number-button button-6",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "6",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "7",
              "className": "number-button button-7",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "7",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "8",
              "className": "number-button button-8",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "8",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "9",
              "className": "number-button button-9",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "9",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "0",
              "className": "number-button button-0",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "0",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login-submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="login-form"
>
            <button
                        className="number-button button-1"
                        onClick={[Function]}
            >
                        1
            </button>
            <button
                        className="number-button button-2"
                        onClick={[Function]}
            >
                        2
            </button>
            <button
                        className="number-button button-3"
                        onClick={[Function]}
            >
                        3
            </button>
            <button
                        className="number-button button-4"
                        onClick={[Function]}
            >
                        4
            </button>
            <button
                        className="number-button button-5"
                        onClick={[Function]}
            >
                        5
            </button>
            <button
                        className="number-button button-6"
                        onClick={[Function]}
            >
                        6
            </button>
            <button
                        className="number-button button-7"
                        onClick={[Function]}
            >
                        7
            </button>
            <button
                        className="number-button button-8"
                        onClick={[Function]}
            >
                        8
            </button>
            <button
                        className="number-button button-9"
                        onClick={[Function]}
            >
                        9
            </button>
            <button
                        className="number-button button-0"
                        onClick={[Function]}
            >
                        0
            </button>
            <button
                        className="login-submit"
                        onClick={[Function]}
            >
                        Login
            </button>
</div>,
          undefined,
        ],
        "className": "login-form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="number-button button-1"
                onClick={[Function]}
>
                1
</button>,
              <button
                className="number-button button-2"
                onClick={[Function]}
>
                2
</button>,
              <button
                className="number-button button-3"
                onClick={[Function]}
>
                3
</button>,
              <button
                className="number-button button-4"
                onClick={[Function]}
>
                4
</button>,
              <button
                className="number-button button-5"
                onClick={[Function]}
>
                5
</button>,
              <button
                className="number-button button-6"
                onClick={[Function]}
>
                6
</button>,
              <button
                className="number-button button-7"
                onClick={[Function]}
>
                7
</button>,
              <button
                className="number-button button-8"
                onClick={[Function]}
>
                8
</button>,
              <button
                className="number-button button-9"
                onClick={[Function]}
>
                9
</button>,
              <button
                className="number-button button-0"
                onClick={[Function]}
>
                0
</button>,
              <button
                className="login-submit"
                onClick={[Function]}
>
                Login
</button>,
            ],
            "className": "login-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "className": "number-button button-1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "1",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "className": "number-button button-2",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "2",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3",
                "className": "number-button button-3",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "3",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "className": "number-button button-4",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "4",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5",
                "className": "number-button button-5",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "5",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6",
                "className": "number-button button-6",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "6",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "7",
                "className": "number-button button-7",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "7",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "8",
                "className": "number-button button-8",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "8",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "9",
                "className": "number-button button-9",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "9",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0",
                "className": "number-button button-0",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "0",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "login-submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
